packages/cli/src/lib/description.ts

```diff
@@ -140,7 +139,7 @@ export class TaskDescription {
   /**
    * Create a new task with initial metadata
    */
-  static create(taskData: CreateTaskData): TaskDescription {
+  static async create(taskData: CreateTaskData): Promise<TaskDescription> {
```

Comment:
Overall, I see many methods are converted to `async` just because they need to discover the project root. I'm considering we can initialize the project root at CLI level (just once) and read it sync after that. We can do it in a CLI hook and initialize a global. 

That will simplify all this migration a lot @ereslibre and will reduce the amount of `async` we need.

Wdyt?

Reply:
Sounds good, I think I will close this PR and reopen with the new approach given there are conflicts now too.